modules/sovrnBidAdapter.js

```diff
@@ -1,156 +1,83 @@
-    adloader.loadScript(scriptUrl);
-  }
-
-  function addBlankBidResponses(impidsWithBidBack) {
-    var missing = utils.getBidderRequestAllAdUnits('sovrn');
-    if (missing) {
-      missing = missing.bids.filter(bid => impidsWithBidBack.indexOf(bid.bidId) < 0);
-    } else {
-      missing = [];
-    }
-
-    missing.forEach(function (bidRequest) {
-      // Add a no-bid response for this bid request.
-      var bid = {};
-      bid = bidfactory.createBid(2, bidRequest);
-      bid.bidderCode = 'sovrn';
-      bidmanager.addBidResponse(bidRequest.placementCode, bid);
-    });
-  }
-
-  // expose the callback to the global object:
-  $$PREBID_GLOBAL$$.sovrnResponse = function (sovrnResponseObj) {
-    var impidsWithBidBack = [];
-
-    // valid response object from sovrn
-    if (sovrnResponseObj && sovrnResponseObj.id && sovrnResponseObj.seatbid && sovrnResponseObj.seatbid.length !== 0 &&
-      sovrnResponseObj.seatbid[0].bid && sovrnResponseObj.seatbid[0].bid.length !== 0) {
-      sovrnResponseObj.seatbid[0].bid.forEach(function (sovrnBid) {
-        var responseCPM;
-        var placementCode = '';
-        var id = sovrnBid.impid;
-        var bid = {};
-
-        var bidObj = utils.getBidRequest(id);
-
-        if (bidObj) {
-          placementCode = bidObj.placementCode;
-          bidObj.status = CONSTANTS.STATUS.GOOD;
-
-          responseCPM = parseFloat(sovrnBid.price);
-
-          if (responseCPM !== 0) {
-            sovrnBid.placementCode = placementCode;
-            sovrnBid.size = bidObj.sizes;
-            var responseAd = sovrnBid.adm;
-
-            // build impression url from response
-            var responseNurl = '<img src="' + sovrnBid.nurl + '">';
-
-            // store bid response
-            // bid status is good (indicating 1)
-            bid = bidfactory.createBid(1, bidObj);
-            bid.creative_id = sovrnBid.id;
-            bid.bidderCode = 'sovrn';
-            bid.cpm = responseCPM;
-
-            // set ad content + impression url
-            // sovrn returns <script> block, so use bid.ad, not bid.adurl
-            bid.ad = decodeURIComponent(responseAd + responseNurl);
-
-            // Set width and height from response now
-            bid.width = parseInt(sovrnBid.w);
-            bid.height = parseInt(sovrnBid.h);
-
-            if (sovrnBid.dealid) {
-              bid.dealId = sovrnBid.dealid;
-            }
-
-            bidmanager.addBidResponse(placementCode, bid);
-            impidsWithBidBack.push(id);
-          }
-        }
+    return {
+      method: 'POST',
+      url: `//ap.lijit.com/rtb/bid?src=${REPO_AND_VERSION}`,
+      data: JSON.stringify(sovrnBidReq),
+      options: {contentType: 'text/plain'}
+    };
+  },
+
+  /**
+   * Format Sovrn responses as Prebid bid responses
+   * @param {id, seatbid} sovrnResponse A successful response from Sovrn.
+   * @return {Bid[]} An array of formatted bids.
+  */
+  interpretResponse: function({id, seatbid}) {
```

Comment:
#1748 changed the first argument of `interpretResponse` to:

```JavaScript
{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
```

so destructuring like
```JavaScript
{body: {id, seatbid}}
```
or however you'd prefer to grab the `body`, and updating corresponding tests should get this back to working properly

Reply:
Thanks @matthewlane for the info here. Just so I am on the same page, destructuring as you have suggested will lead to the same functionality as what we currently have implemented? I am just making sure that we do not have to change the bid response's JSON structure on our end.

Reply:
That's right, updating in this way will give the same functionality as before #1748 was merged to master, and there are no changes required to the JSON structure sent from your endpoint.

For a (non-destructuring) example, appnexusAst was updated [here](https://github.com/prebid/Prebid.js/blob/47019481bb34f4071f85d899a16727d8aae606e0/modules/appnexusAstBidAdapter.js#L90) to pick the response data off the `body` property. Before #1748 we used the first parameter directly. This update lets us keep working properly and no changes were made to our endpoint because behind the scenes it's the same data coming through, just on the `body` property of the first parameter rather than the first parameter itself now.

Let me know if you have more questions, and I'll verify the change works whenever the PR is updated